### PR TITLE
feat: Adds image attachment for help center articles

### PR DESCRIPTION
**Pull request by nithindavid**
## Description

Users will be able to add images to their articles by clicking the "Insert Image" button in the editor toolbar. The image will be uploaded and inserted into the article at the cursor position.

The image will be displayed within the article, and users will be able to add a caption for the image as needed. 

Fixes # (issue)

Loom: https://www.loom.com/share/31d1a175674044bbb7ef350b29431718

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by iamsivin — COMMENTED** [review]
@nithindavid I pulled this branch in locally for testing. Not able to see the image upload icon.
image

**Comment by fayazara** [comment]
> @nithindavid I pulled this branch in locally for testing. Not able to see the image upload icon. image

@iamsivin Do this in package.json and run yarn

Screenshot2023-02-09 at 17 05 30@2x

@nithindavid Can I update this?

**Review by fayazara — COMMENTED** [review]
This works really well 🔥🔥

**Tested -** 
1. Uploading PNGs, JPGs, Webp, GIFs works - ✅
2. Is responsive - ✅

**Found a small bug -** 
When I upload a new image in the editor, the `Publish button` stays disabled, works only when it's a new article and I have added some text in it and the button has been activated.

**Questions.**
Should we add the line `Add a caption for the image`, since this is not a input placeholder, someone might not see it and this same line could go published (happened to me while testing)

**Review by iamsivin — APPROVED** [review]
I tested it locally and LGTM 💯
https://www.loom.com/share/ca4c990839aa406c899b94c1105b5cf0

@nithindavid Do we support SVG? Currently, I can select SVG from the file selector, and the image is broken in the article editor UI. You can see it in the below screen recording.

https://user-images.githubusercontent.com/64252451/217842408-efb3b672-f10e-404a-908a-ba24380eae97.mov

**Comment by pranavrajs** [comment]
@iamsivin Let us disable SVG upload as in other places.

**Comment by nithindavid** [comment]
@iamsivin @fayazara I have removed the caption altogether. 
Small images, they would be still rendered inline. Users can break to the next line to keep the text looking clean.


@fayazara The publish button issue is a bug, since unrelated we can create a separate PR. @iamsivin Can you check this?
 Can you check?

File: app/javascript/dashboard/components/widgets/WootWriter/FullEditor.vue [hunk 1]
```diff
@@ -1,6 +1,13 @@
 <template>
   <div>
     <div class="editor-root editor--article">
+      <input
+        ref="imageUploadInput"
+        type="file"
+        accept="image/png, image/jpeg, image/jpg, image/gif"
```
Comment (scmmishra) [inline app/javascript/dashboard/components/widgets/WootWriter/FullEditor.vue:7]:
```suggestion
        accept="image/*"
```

We could use this to accept all kinds of images, unless we're limiting to these for a specific reason. Or we could add `image/webp`

File: app/javascript/dashboard/components/widgets/WootWriter/FullEditor.vue [hunk 2]
```diff
@@ -16,23 +23,31 @@ import {
...
   Selection,
 } from '@chatwoot/prosemirror-schema';
-
+import { checkFileSizeLimit } from 'shared/helpers/FileHelper';
+import alertMixin from 'shared/mixins/alertMixin';
 import eventListenerMixins from 'shared/mixins/eventListenerMixins';
 import uiSettingsMixin from 'dashboard/mixins/uiSettings';
 
-const createState = (content, placeholder, plugins = []) => {
+const MAXIMUM_FILE_UPLOAD_SIZE = 16; // in MB
```
Comment (scmmishra) [inline app/javascript/dashboard/components/widgets/WootWriter/FullEditor.vue:31]: Should we reduce this to 4 or 5 MB? Allowing these file sizes is not good for either our storage costs neither loading speed on their web pages
```suggestion
const MAXIMUM_FILE_UPLOAD_SIZE = 4; // in MB
```

File: app/javascript/dashboard/components/widgets/WootWriter/FullEditor.vue [hunk 3]
```diff
@@ -73,8 +93,77 @@ export default {
...
+            )
+          );
+
+        this.editorView.dispatch(tr.scrollIntoView());
+        this.focusEditorInputField();
+      }
+    },
+    onImageUploadStop() {
+      // append the image to the editor
+    },
```
Comment (scmmishra) [inline app/javascript/dashboard/components/widgets/WootWriter/FullEditor.vue:159]: Should we remove this method if it's not used

**Review by scmmishra — CHANGES_REQUESTED** [review]

File: app/javascript/dashboard/components/widgets/WootWriter/FullEditor.vue (hunk 1 above)
Comment (scmmishra) [reply app/javascript/dashboard/components/widgets/WootWriter/FullEditor.vue:7]: I see in the previous conversation, SVG support is broken. Let's just add `image/webp` then. Updating suggestion

Comment (iamsivin) [reply app/javascript/dashboard/components/widgets/WootWriter/FullEditor.vue:7]: @scmmishra I added it in this PR. The reason is if we add an SVG in the article editor, It looks broken. You can see it in this video.
https://user-images.githubusercontent.com/64252451/217842408-efb3b672-f10e-404a-908a-ba24380eae97.mov

Comment (scmmishra) [reply app/javascript/dashboard/components/widgets/WootWriter/FullEditor.vue:7]: Noted, let's just allow webp too

File: app/javascript/dashboard/components/widgets/WootWriter/FullEditor.vue (hunk 2 above)
Comment (nithindavid) [reply app/javascript/dashboard/components/widgets/WootWriter/FullEditor.vue:31]: Makes sense

**Comment by nithindavid** [comment]
@iamsivin @scmmishra Thanks for the reviews 🙌 I have made the changes

**Comment by pranavrajs** [comment]
@nithindavid I'm not sure if there is any setup issue, but I cannot test this locally. The image upload button is not available.

**Review by pranavrajs — CHANGES_REQUESTED** [review]
Please see the comment above.

**Comment by iamsivin** [comment]
> @nithindavid I'm not sure if there is any setup issue, but I cannot test this locally. The image upload button is not available.

@pranavrajs I think you can follow these steps for testing, WFM https://github.com/chatwoot/chatwoot/pull/6426#issuecomment-1424046582

**Comment by pranavrajs** [comment]
@nithindavid / @iamsivin There are multiple issues with this package. 

1. The main branch is pointed at #variable-mention. This PR points to a different one. However, the changes in the variable-mention branch don't seem to be available in the feat-image-upload-beta.
2. Even with the feat-image-upload-beta, the yarn.lock file points to an earlier commit. Is that intentional?

**Comment by nithindavid** [comment]
@pranavrajs I have updated the npm package hash. Please check now

**Review by pranavrajs — APPROVED** [review]

**Comment by iamkhalidbashir** [comment]
Does this delete images if no longer used in an article?

**Pull request merged**